base/applications/screensavers/logon/logon.c

```diff
@@ -18,162 +18,112 @@
-                    bm.bmHeight,
-                    SRCCOPY);
+                StretchBlt(hdc, RANDOM (0, rect.right - (bm.bmWidth / 5)),
+                           RANDOM (0, rect.bottom - (bm.bmHeight / 5)),
+                           bm.bmWidth / 5, bm.bmHeight / 5, hdcMem, 0, 0,
+                           bm.bmWidth, bm.bmHeight, SRCCOPY);
              }
              else
              {
-                 BitBlt(
-                     hdc,
-                     RANDOM (0, rect.right - bm.bmWidth),
-                     RANDOM (0, rect.bottom - bm.bmHeight),
-                     bm.bmWidth,
-                     bm.bmHeight,
-                     hdcMem,
-                     0,
-                     0,
-                     SRCCOPY);
+                 BitBlt(hdc, RANDOM (0, rect.right - bm.bmWidth),
+                        RANDOM (0, rect.bottom - bm.bmHeight),
+                        bm.bmWidth, bm.bmHeight, hdcMem, 0, 0, SRCCOPY);
              }
 
              SelectObject(hdcMem, hbmOld);
              DeleteDC(hdcMem);
-
              EndPaint(hWnd, &ps);
              break;
         }
         case WM_TIMER:
-        {
-          InvalidateRect(hWnd, NULL, 1);
-          break;
-        }
+            InvalidateRect(hWnd, NULL, 1);
+            break;
+
         case WM_DESTROY:
-        {
             KillTimer(hWnd, APP_TIMER);
             DeleteObject(bitmap);
             PostQuitMessage(0);
             break;
-        }
 
         default:
-            // Pass Windows Messages to the default screensaver window procedure
-            return DefScreenSaverProc(hWnd, message, wParam, lParam);
+            /* Pass Windows messages to the default screensaver window procedure */
+            return DefScreenSaverProc(hWnd, uMsg, wParam, lParam);
     }
 
     return 0;
 }
 
-BOOL
-WINAPI
-ScreenSaverConfigureDialog(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
+BOOL WINAPI ScreenSaverConfigureDialog(HWND hDlg, UINT uMsg, WPARAM wParam, LPARAM lParam)
 {
     return FALSE;
 }
 
-// This function is only called one time before opening the configuration dialog.
-// Use it to show a message that no configuration is necessary and return FALSE to indicate that no configuration dialog shall be opened.
-BOOL
-WINAPI
-RegisterDialogClasses(HANDLE hInst)
+/* This function is only called one time before opening the configuration dialog.
+ * Use it to show a message that no configuration is necessary and return FALSE to indicate that no configuration dialog shall be opened.
+ */
+BOOL WINAPI RegisterDialogClasses(HANDLE hInst)
 {
-    TCHAR szMessage[256];
-    TCHAR szTitle[25];
+    WCHAR szMessage[256];
+    WCHAR szTitle[25];
 
-    LoadString(hInst, IDS_TEXT, szMessage, sizeof(szMessage) / sizeof(TCHAR));
-    LoadString(hInst, IDS_DESCRIPTION, szTitle, sizeof(szTitle) / sizeof(TCHAR));
+    LoadString(hInst, IDS_TEXT, szMessage, wcslen(szMessage));
+    LoadString(hInst, IDS_DESCRIPTION, szTitle, wcslen(szTitle));
```

Comment:
No, this won't work.
If those buffers in the function are initialized with NULLs, wcslen() will returrn 0.

Reply:
Reverted.